pkg/confidence/confidence.go

```diff
@@ -57,6 +57,9 @@ type ConfidenceBuilder struct {
 
 func (e ConfidenceBuilder) SetAPIConfig(config APIConfig) ConfidenceBuilder {
 	e.confidence.Config = config
+	if config.APIResolveBaseUrl == "" {
+		e.confidence.Config.APIResolveBaseUrl = DefaultAPIResolveBaseUrl
+	}
```

Comment:
Added this to protect against this case:
```
c.NewConfidenceBuilder().SetAPIConfig(c.APIConfig{APIKey: "API_KEY"}).Build()
```
Instantiating the APIConfig{} struct directly can lead to baseUrl being an empty string